### PR TITLE
ENT-2468: Use format strings with named arguments for translator context

### PR DESCRIPTION
**Pull request by tlhmerry9**

### CODE REVIEW

File: src/rct/cert_commands.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def _create_cert(self):
             return certificate.create_from_file(cert_file)
         except certificate.CertificateException as ce:
             raise InvalidCLIOptionError(
-                    _("Unable to read certificate file '%s': %s") % (cert_file,
-                        ce))
+                    _("Unable to read certificate file '{cert}}': {ce}}".format(cert=cert_file,
```
Comment (ptoscano) [inline src/rct/cert_commands.py:46]: extra `}` twice

File: src/rhsmlib/facts/dmiinfo.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def _read_dmi(self, func):
         try:
             return func()
         except Exception as e:
-            log.warn(_("Error reading system DMI information with %s: %s"), func, e)
+            log.warn(_("Error reading system DMI information with {func}}: {e}}".format(func=func, e=e)))
```
Comment (ptoscano) [inline src/rhsmlib/facts/dmiinfo.py:100]: extra `}` twice

**Review by ptoscano — CHANGES_REQUESTED** [review]
There is a common mistake done in all the conversions: you have to apply `format()` on the result of `_()`, not on the original string, otherwise they do not get translated.

File: src/rct/cert_commands.py (hunk 1 above)
Comment (tlhmerry9) [reply src/rct/cert_commands.py:46]: removed extra `}`

File: src/rct/manifest_commands.py [hunk 3]
```diff
@@ -267,8 +267,8 @@ def _print_products(self, zip_archive):
                 cert = certificate.create_from_pem(zip_archive._read_file(cert_file).decode('utf-8'))
             except certificate.CertificateException as ce:
                 raise certificate.CertificateException(
-                        _("Unable to read certificate file '%s': %s") % (cert_file,
-                        ce))
+                        _("Unable to read certificate file '{cert}': {ce}}".format(cert=cert_file,
```
[the remark this answers is not in the document]
Comment (tlhmerry9) [reply src/rct/manifest_commands.py:270]: removed extra `}`

File: src/rhsmlib/facts/dmiinfo.py (hunk 2 above)
Comment (tlhmerry9) [reply src/rhsmlib/facts/dmiinfo.py:100]: removed extra `}`

**Comment by ptoscano** [comment]
This also needs to be rebased on top of current master, without merge commits.

**Comment by tlhmerry9** [comment]
> There is a common mistake done in all the conversions: you have to apply `format()` on the result of `_()`, not on the original string, otherwise they do not get translated.

I updated the way I added `.format()` to be `_().format()` and I also went through these files and replace the majority of the `"%s" % value`.  I left a couple and will be looking at them in a few minutes.

File: src/subscription_manager/i18n_optparse.py [hunk 4]
```diff
@@ -143,5 +143,5 @@ def error(self, msg):
         """
         self.print_usage(sys.stderr)
         #translators: arg 1 is the program name, arg 2 is the error message
-        print(_("%s: error: %s") % (self.get_prog_name(), msg))
+        print(_("{name}}: error: {msg}}".format(name=self.get_prog_name(), msg=msg)))
```
[the remark this answers is not in the document]
Comment (tlhmerry9) [reply src/subscription_manager/i18n_optparse.py]: fixed but this file was deleted and will not be included in this PR

**Comment by tlhmerry9** [comment]
> This also needs to be rebased on top of current master, without merge commits.

This has been done and squashed to one commit

**Comment by tlhmerry9** [comment]
> This also needs to be rebased on top of current master, without merge commits.

I need to fix some tests before you re-review

File: src/rhsm_debug/debug_commands.py [hunk 5]
```diff
@@ -181,13 +182,13 @@ def _do_command(self):
...
 
-                final_path = os.path.join(self.options.destination, "rhsm-debug-system-%s.tar.gz" % code)
+                final_path = os.path.join(self.options.destination, "rhsm-debug-system-{code}.tar.gz".format(code=code))
 
                 self.final_destination_path = final_path
 
                 sfm = SaferFileMove()
                 sfm.move(tar_file_path, final_path)
-                print(_("Wrote: %s") % final_path)
+                print(_("Wrote: {final_path}\\").format(final_path=final_path))
```
Comment (ptoscano) [inline src/rhsm_debug/debug_commands.py:191]: extra `\\`

File: src/subscription_manager/cache.py [hunk 6]
```diff
@@ -231,7 +231,7 @@ def load_status(self, uep, uuid, on_date=None):
             self.last_error = ex
             log.error("Bad identity, unable to connect to server")
             return None
-        # all of the abover are subclasses of ConnectionException that
+        # all of the above are subclasses of ConnectionException that
```
Comment (ptoscano) [inline src/subscription_manager/cache.py:234]: while it's a good typo fix, i'd do it in a separate PR (usually better avoid unrelated changes)

File: src/subscription_manager/cache.py [hunk 7]
```diff
@@ -424,7 +424,7 @@ def profile_reporting_enabled(self):
         # If the environment variable is 0, defer to the setting in the conf file; likewise if the environment
         # variable is completely unset.
         if 'SUBMAN_DISABLE_PROFILE_REPORTING' in os.environ and \
-            os.environ['SUBMAN_DISABLE_PROFILE_REPORTING'].lower() in ['true', '1', 'yes', 'on']:
+                os.environ['SUBMAN_DISABLE_PROFILE_REPORTING'].lower() in ['true', '1', 'yes', 'on']:
```
Comment (ptoscano) [inline src/subscription_manager/cache.py:427]: unrelated change

File: src/subscription_manager/cache.py [hunk 8]
```diff
@@ -706,10 +706,8 @@ def check_for_update(self):
                 data = json.loads(self.read())
                 last_update = parse_date(data["lastUpdate"])
             except (ValueError, KeyError) as err:
-                log.debug("Cache file {file} is corrupted: {err}".format(
-                    file=self.CACHE_FILE,
-                    err=err
-                ))
+                log.debug("Cache file {c_file} is corrupted: {err}".format(c_file=self.CACHE_FILE,
+                                                                           err=err))
```
Comment (ptoscano) [inline src/subscription_manager/cache.py:710]: unrelated (and unneeded IMHO) change

File: src/subscription_manager/gui/about.py [hunk 9]
```diff
@@ -73,10 +73,10 @@ def __init__(self, parent, backend):
         server_versions = get_server_versions(self.backend.cp_provider.get_consumer_auth_cp())
         client_versions = get_client_versions()
 
-        sub_man_version_label.set_markup(_("<b>%s version:</b> %s") %
-                                        ("subscription manager", client_versions['subscription-manager']))
-        backend_version_label.set_markup(_("<b>subscription management service version:</b> %s") %
-                                           server_versions['candlepin'])
+        sub_man_version_label.set_markup(_("<b>{name} version:</b> {version}").format(
+            name="subscription manager", version=client_versions['subscription-manager']))
+        backend_version_label.set_markup(_("<b>subscription management service {version}:</b> {version}").format(
```
Comment (ptoscano) [inline src/subscription_manager/gui/about.py:78]: the first `version` is just a word, no need to replace it

File: src/subscription_manager/gui/about.py [hunk 10]
```diff
@@ -119,7 +119,6 @@ def _rhsmcertd_on(self):
         fnull = open(os.devnull, "w")
         try:
             # if status == 0 then true
-
```
Comment (ptoscano) [inline src/subscription_manager/gui/about.py:122]: extra change

File: src/subscription_manager/migrate/migrate.py [hunk 11]
```diff
@@ -385,12 +387,13 @@ def check_has_access(self, rpc_session, session_key):
                 # We should not ever be here.  This method has a guard that keeps it from being
                 # called when not needed.  If we see this error, someone has made a programming
                 # mistake.
-                raise Exception("No session key available.  Check that XMLRPC connection is being made with credentials.")
+                raise Exception("No session key available.  "
+                                "Check that XMLRPC connection is being made with credentials.")
```
Comment (ptoscano) [inline src/subscription_manager/migrate/migrate.py:391]: unrelated/unneeded change

File: src/subscription_manager/migrate/migrate.py [hunk 12]
```diff
@@ -945,13 +954,15 @@ def validate_options(options):
...
+            system_exit(os.EX_USAGE,
+                        _("The --activation-key option precludes the use of --destination-user and "
+                          " --destination-password"))
         if not options.org:
             system_exit(os.EX_USAGE, _("The --activation-key option requires that a --org be given."))
 
     if options.service_level and not options.auto:
         # TODO Need to explain why this restriction exists.
-        system_exit(os.EX_USAGE, _("The --servicelevel and --no-auto options cannot be used together."))
+        system_exit(os.EX_USAGE, _("The --service-level and --no-auto options cannot be used together."))
```
Comment (ptoscano) [inline src/subscription_manager/migrate/migrate.py:965]: good change, however unrelated; better do it in an own PR

**Review by ptoscano — CHANGES_REQUESTED** [review]
This is a good improvements, thanks. However, I have few notes:
- I'd avoid updating `keys.pot` in the same commit, as it makes the patches bigger; also, the updated pot files has few strings less, so I'd investigate that as a separate task
- adding named placeholders in translatable strings is a good thing; however, I'd be nice to actually have meaningful names for the placeholders when possible: instead of `e`, `ec`, `msg`, `fn`, `c_file`, etc, I'd use less shortened forms like `error`, `exception`, `message`, `function`, `command`, `certificate`, etc
- other than translatable strings, also helper strings were changed; I personally consider this a different task/story than changing UI strings, as IMHO code like `"%s.json" % name` is still better than the `.format()` syntax (and there is no need to reorder placeholders in non-UI strings)

File: src/rhsm_debug/debug_commands.py (hunk 5 above)
Comment (tlhmerry9) [reply src/rhsm_debug/debug_commands.py:191]: This was a typo!  Fixed

File: src/subscription_manager/cache.py (hunk 7 above)
Comment (tlhmerry9) [reply src/subscription_manager/cache.py:427]: changed back

File: src/subscription_manager/gui/about.py (hunk 9 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/about.py:78]: I think I was thinking I forgot the {} around it.  changed it back

File: src/subscription_manager/gui/about.py (hunk 10 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/about.py:122]: put back the extra line

File: src/subscription_manager/migrate/migrate.py (hunk 12 above)
Comment (tlhmerry9) [reply src/subscription_manager/migrate/migrate.py:965]: Changed back and made a note so I can open a PR

File: src/subscription_manager/migrate/migrate.py (hunk 11 above)
Comment (tlhmerry9) [reply src/subscription_manager/migrate/migrate.py:391]: put back

File: src/subscription_manager/cache.py (hunk 6 above)
Comment (tlhmerry9) [reply src/subscription_manager/cache.py:234]: put back and made note to open a PR, and going to include the typo in migrate.py also

File: src/subscription_manager/cache.py (hunk 8 above)
Comment (tlhmerry9) [reply src/subscription_manager/cache.py:710]: put back

**Comment by tlhmerry9** [comment]
> This is a good improvements, thanks. However, I have few notes:
> 
> * I'd avoid updating `keys.pot` in the same commit, as it makes the patches bigger; also, the updated pot files has few strings less, so I'd investigate that as a separate task
> * adding named placeholders in translatable strings is a good thing; however, I'd be nice to actually have meaningful names for the placeholders when possible: instead of `e`, `ec`, `msg`, `fn`, `c_file`, etc, I'd use less shortened forms like `error`, `exception`, `message`, `function`, `command`, `certificate`, etc
> * other than translatable strings, also helper strings were changed; I personally consider this a different task/story than changing UI strings, as IMHO code like `"%s.json" % name` is still better than the `.format()` syntax (and there is no need to reorder placeholders in non-UI strings)

> using a restored `keys.pot` file
> made the names used in the format strings to be more descriptive
> put the other non-translatable strings back the way they were (using %s)

File: src/subscription_manager/gui/registergui.py [hunk 13]
```diff
@@ -1523,10 +1523,10 @@ def __init__(self, reg_info, async_backend, parent_window):
 
         self._initialize_consumer_name()
         self.registration_tip_label.set_label("<small>%s</small>" %
-                                          get_branding().GUI_FORGOT_LOGIN_TIP)
+                                              get_branding().GUI_FORGOT_LOGIN_TIP)
```
Comment (ptoscano) [inline src/subscription_manager/gui/registergui.py:1526]: extra indentation change

File: src/subscription_manager/gui/registergui.py [hunk 14]
```diff
@@ -1523,10 +1523,10 @@ def __init__(self, reg_info, async_backend, parent_window):
...
 
         self.registration_header_label.set_label("<b>%s</b>" %
-                                             get_branding().GUI_REGISTRATION_HEADER)
+                                                 get_branding().GUI_REGISTRATION_HEADER)
```
Comment (ptoscano) [inline src/subscription_manager/gui/registergui.py:1529]: extra indentation change

File: src/subscription_manager/gui/registergui.py [hunk 15]
```diff
@@ -1748,8 +1748,8 @@ def _on_default_button_clicked(self, widget):
         # Default port and prefix are fine, so we can be concise and just
         # put the hostname for RHN:
         self.server_entry.set_text("%s:%s%s" % (base_config.DEFAULT_HOSTNAME,
-            base_config.DEFAULT_PORT,
-            base_config.DEFAULT_PREFIX))
+                                                base_config.DEFAULT_PORT,
+                                                base_config.DEFAULT_PREFIX))
```
Comment (ptoscano) [inline src/subscription_manager/gui/registergui.py:1752]: extra indentation changes

File: src/subscription_manager/gui/registergui.py [hunk 16]
```diff
@@ -1805,7 +1805,7 @@ def set_server_entry(self, hostname, port, prefix):
             self.server_entry.set_text(base_config.DEFAULT_HOSTNAME)
         else:
             self.server_entry.set_text("%s:%s%s" % (hostname,
-                                       port, prefix))
+                                                    port, prefix))
```
Comment (ptoscano) [inline src/subscription_manager/gui/registergui.py:1808]: extra indentation change

File: src/subscription_manager/gui/registergui.py [hunk 17]
```diff
@@ -2049,7 +2048,7 @@ def _subscribe(self, uuid, current_sla, dry_run_result, callback):
         """
         try:
             log.debug("Binding to subscriptions at service level: %s" %
-                    dry_run_result.service_level)
+                      dry_run_result.service_level)
```
Comment (ptoscano) [inline src/subscription_manager/gui/registergui.py:2051]: extra indentation change

File: src/subscription_manager/migrate/migrate.py [hunk 18]
```diff
@@ -945,7 +950,9 @@ def validate_options(options):
         if options.environment:
             system_exit(os.EX_USAGE, _("The --activation-key and --environment options cannot be used together."))
         if options.destination_user or options.destination_password:
-            system_exit(os.EX_USAGE, _("The --activation-key option precludes the use of --destination-user and --destination-password"))
+            system_exit(os.EX_USAGE,
+                        _("The --activation-key option precludes the use of --destination-user and "
+                          " --destination-password"))
```
Comment (ptoscano) [inline src/subscription_manager/migrate/migrate.py:955]: extra indentation change

**Review by ptoscano — CHANGES_REQUESTED** [review]
Looking good now! There is just one placeholder to fix, and few extra indentation changes to rollback.

File: src/rct/cert_commands.py [hunk 19]
```diff
@@ -86,19 +86,19 @@ def __init__(self):
     def _do_command(self):
         cert = self._create_cert()
         pem = self._get_pem(self._get_file_from_args())
-        print(_("Type: %s") % type_to_string(cert))
-        print(_("Version: %s") % cert.version)
-        print(_("DER size: %db") % self._get_der_size(pem))
+        print(_("Type: {certificate_type}").format(certificate_type=type_to_string(cert)))
+        print(_("Version: {certificate_version}").format(certificate_version=cert.version))
+        print(_("DER size: {open_file}b").format(open_file=self._get_der_size(pem)))
```
[the remark this answers is not in the document]
Comment (tlhmerry9) [reply src/rct/cert_commands.py:91]: reverted back

File: src/subscription_manager/gui/registergui.py (hunk 13 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/registergui.py:1526]: Done

File: src/subscription_manager/gui/registergui.py (hunk 14 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/registergui.py:1529]: done

File: src/subscription_manager/gui/registergui.py (hunk 15 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/registergui.py:1752]: Done

File: src/subscription_manager/gui/registergui.py (hunk 16 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/registergui.py:1808]: Done

File: src/subscription_manager/gui/registergui.py (hunk 17 above)
Comment (tlhmerry9) [reply src/subscription_manager/gui/registergui.py:2051]: Done

File: src/subscription_manager/migrate/migrate.py (hunk 18 above)
Comment (tlhmerry9) [reply src/subscription_manager/migrate/migrate.py:955]: Done

**Comment by tlhmerry9** [comment]
> Looking good now! There is just one placeholder to fix, and few extra indentation changes to rollback.

I made all of the changes, please review again :)

File: src/rct/cert_commands.py [hunk 20]
```diff
@@ -86,19 +86,19 @@ def __init__(self):
     def _do_command(self):
         cert = self._create_cert()
         pem = self._get_pem(self._get_file_from_args())
-        print(_("Type: %s") % type_to_string(cert))
-        print(_("Version: %s") % cert.version)
+        print(_("Type: {certificate_type}").format(certificate_type=type_to_string(cert)))
+        print(_("Version: {certificate_version}").format(certificate_version=cert.version))
         print(_("DER size: %db") % self._get_der_size(pem))
```
Comment (ptoscano) [inline src/rct/cert_commands.py:91]: this was fine to change btw, it needs a `{size}` or similar placeholder

File: src/subscription_manager/migrate/migrate.py [hunk 21]
```diff
@@ -945,7 +950,7 @@ def validate_options(options):
         if options.environment:
             system_exit(os.EX_USAGE, _("The --activation-key and --environment options cannot be used together."))
         if options.destination_user or options.destination_password:
-            system_exit(os.EX_USAGE, _("The --activation-key option precludes the use of --destination-user and --destination-password"))
+            system_exit(os.EX_USAGE, _("The --activation-key option precludes the use of --destination-user and  --destination-password"))
```
Comment (ptoscano) [inline src/subscription_manager/migrate/migrate.py:953]: extra whitespace addition

**Review by ptoscano — CHANGES_REQUESTED** [review]
:+1: very nice improvements, almost there

File: src/rct/cert_commands.py (hunk 20 above)
Comment (tlhmerry9) [reply src/rct/cert_commands.py:91]: I changed it back to use format with a `{size}` placeholder

**Comment by ptoscano** [comment]
:+1: LGTM now -- thanks for the work and the patience!

**Review by ptoscano — APPROVED** [review]
